### ISSUE TITLE
check_disk - exclude squashfs, etc. by default

### ISSUE THREAD

**Issue by bryceharrington** [monitoring-plugins/monitoring-plugins#1608]
In Ubuntu, squashfs is routinely used for snap files, but since these are synthetic filesystems they display as 100% utilization and thus trigger check_disk to issue a false positive DISK CRITICAL error.  

Similar false positives have been reported to us for tracefs and tmpfs. The commonly discussed workaround is to exclude these when running the tools (e.g. using the '-X <fs>' parameter for check_all_disks). Since wrappers are typically used for running the underlying tools, it is possible to add a string of -X... parameters.

Unfortunately the number of flags needed is getting extensive, and I wonder if this could be solved more uniformly.  For filesystems that are always going to show 100% capacity, a cleaner solution is possible. monitoring-plugins' check_disk.c maintains an internal exclusion list, fs_exclude_list, which already excludes iso9660, and can be modified to add other filesystems to exclude by default.

In other words, check_disk.c would be modified thusly:

  np_add_name(&fs_exclude_list, "iso9660");
  np_add_name(&fs_exclude_list, "squashfs");
  np_add_name(&fs_exclude_list, "tmpfs");
  np_add_name(&fs_exclude_list, "tracefs");

This code is added prior to the command line parsing logic, and as such simply sets default behavior. It does not preclude further adding or removing filesystems via the -X and -N parameters. Indeed, if someone were to desire checking tmpfs, they are able to manually add it, via "-N tmpfs".

**Comment by odenbach** [comment]
I personally do not care if I have to add 12 or 16 parameters to check_disk, as that is done once in the Icinga2 command definition. I already exclude

		"none",
		"tmpfs",
		"sysfs",
		"proc",
		"configfs",
		"devtmpfs",
		"devfs",
		"mtmfs",
		"tracefs",
		"cgroup",
		"fuse.gvfsd-fuse",
		"fuse.gvfs-fuse-daemon",
		"fdescfs",
		"overlay",
		"nsfs",
		"squashfs"

But that is of course site-dependant.

**Comment by NHellFire** [comment]
I'm voting no on adding tmpfs to the exclude list, as excludes can't be overridden by command line arguments. tmpfs also reports correctly on my Ubuntu systems:
```
nathan@nagios:~$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.5 LTS"
nathan@nagios:~$ uname -r
4.15.0-1067-raspi2
nathan@nagios:~$ df -t tmpfs -h
Filesystem      Size  Used Avail Use% Mounted on
tmpfs            93M  4.3M   89M   5% /run
tmpfs           462M     0  462M   0% /dev/shm
tmpfs           5.0M  4.0K  5.0M   1% /run/lock
tmpfs           462M     0  462M   0% /sys/fs/cgroup
tmpfs            93M     0   93M   0% /run/user/10001
```

**Comment by RincewindsHat** [comment]
I think, this is better left over to the monitoring system, not the plugin
